internal/service/grafana/workspace_api_key_test.go

```diff
@@ -0,0 +1,70 @@
+package grafana_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/managedgrafana"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func testAccWorkspaceAPIKey_basic(t *testing.T) {
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_grafana_workspace_api_key.test"
+	workspaceResourceName := "aws_grafana_workspace.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t); acctest.PreCheckPartitionHasService(managedgrafana.EndpointsID, t) },
+		ErrorCheck:        acctest.ErrorCheck(t, managedgrafana.EndpointsID),
+		CheckDestroy:      nil,
+		ProviderFactories: acctest.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWorkspaceAPIKeyConfig_basic(rName, "test-api-1"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "key_name", "test-api-1"),
+					resource.TestCheckResourceAttr(resourceName, "key_role", "EDITIR"),
+					resource.TestCheckResourceAttr(resourceName, "seconds_to_live", "3600"),
+					resource.TestCheckResourceAttrPair(resourceName, "workspace_id", workspaceResourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "key"),
+				),
+			},
+		},
+	})
+}
+
+func testAccWorkspaceAPIKeyConfig_basic(rName string, apiKey string) string {
```

Comment:
For this test function, 2 parameters are not required, we should be using 1 parameter since only 1 parameter is used in the hcl definition

Reply:
removed the extra parameter.